integrationtests/src/test/java/org/axonframework/integrationtests/eventhandling/TrackingEventProcessorTest.java

```diff
@@ -753,15 +771,323 @@ public void testHasAvailableSegments() {
         assertWithin(2, TimeUnit.SECONDS, () -> assertEquals(1, testSubject.availableProcessorThreads()));
     }
 
+    @Test(timeout = 10000)
+    public void testSplitSegments() throws InterruptedException {
+        tokenStore.initializeTokenSegments(testSubject.getName(), 1);
+        testSubject.start();
+        waitForSegmentStart(0);
+        assertTrue("Expected split to succeed", testSubject.splitSegment(0).join());
+        assertArrayEquals(new int[]{0, 1}, tokenStore.fetchSegments(testSubject.getName()));
+        waitForSegmentStart(0);
+    }
+
+    @Test(timeout = 10000)
+    public void testMergeSegments() throws InterruptedException {
+        tokenStore.initializeTokenSegments(testSubject.getName(), 2);
+        testSubject.start();
+        while (testSubject.processingStatus().isEmpty()) {
+            Thread.sleep(10);
+        }
+
+        assertTrue("Expected merge to succeed", testSubject.mergeSegment(0).join());
+        assertArrayEquals(new int[]{0}, tokenStore.fetchSegments(testSubject.getName()));
+        waitForSegmentStart(0);
+    }
+
+    @Test(timeout = 10000)
+    public void testMergeSegments_BothClaimedByProcessor() throws Exception {
+        initProcessor(TrackingEventProcessorConfiguration.forParallelProcessing(2));
+        tokenStore.initializeTokenSegments(testSubject.getName(), 2);
+        List<EventMessage<?>> handledEvents = new CopyOnWriteArrayList<>();
```

Comment:
It's a list, exactly for that reason. If it's handled twice, we'd have to be able to record it twice, so that we can detect the duplicate. Note that the "same" event doesn't mean "equal objects" as the  tracking token they appear with may be different. They will share the same message ID, though.
Added assertions to ensure no duplicate delivery has taken place.